Mage/src/main/java/mage/abilities/effects/common/search/SearchLibraryPutOneOntoBattlefieldTappedRestInHandEffect.java

```diff
@@ -0,0 +1,79 @@
+package mage.abilities.effects.common.search;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.SearchEffect;
+import mage.cards.Card;
+import mage.cards.Cards;
+import mage.cards.CardsImpl;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.target.common.TargetCardInLibrary;
+
+import java.util.Set;
+
+/**
+ * @author BetaSteward_at_googlemail.com, edited by Cguy7777
+ */
+public class SearchLibraryPutOneOntoBattlefieldTappedRestInHandEffect extends SearchEffect {
+
+    private static final FilterCard filter = new FilterCard("card to put on the battlefield tapped");
+
+    public SearchLibraryPutOneOntoBattlefieldTappedRestInHandEffect(TargetCardInLibrary target) {
+        super(target, Outcome.PutLandInPlay);
+        staticText = "search your library for " + target.getDescription() +
+                ", reveal those cards, put one onto the battlefield tapped and the other into your hand, then shuffle";
+    }
+
+    protected SearchLibraryPutOneOntoBattlefieldTappedRestInHandEffect(final SearchLibraryPutOneOntoBattlefieldTappedRestInHandEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public SearchLibraryPutOneOntoBattlefieldTappedRestInHandEffect copy() {
+        return new SearchLibraryPutOneOntoBattlefieldTappedRestInHandEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        MageObject sourceObject = game.getObject(source);
+        if (controller == null || sourceObject == null) {
+            return false;
+        }
+
+        if (controller.searchLibrary(target, source, game)) {
+            if (!target.getTargets().isEmpty()) {
+                Cards revealed = new CardsImpl(target.getTargets());
+                controller.revealCards(sourceObject.getIdName(), revealed, game);
+
+                if (target.getTargets().size() >= 2) {
+                    TargetCard targetCardToBattlefield = new TargetCard(Zone.LIBRARY, filter);
+                    controller.choose(Outcome.PutLandInPlay, revealed, targetCardToBattlefield, source, game);
+                    Card cardToBattlefield = revealed.get(targetCardToBattlefield.getFirstTarget(), game);
+                    if (cardToBattlefield != null) {
+                        controller.moveCards(cardToBattlefield, Zone.BATTLEFIELD, source, game, true, false, false, null);
+                        revealed.remove(cardToBattlefield);
```

Comment:
It's there so that the card we just put onto the battlefield isn't moved to your hand along with the other cards later. I've rewritten this area a bit to try to make it easier to understand.